src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -195,4 +204,92 @@ public AssemblyService setCacheSize(int CACHE_SIZE) {
         this.CACHE_SIZE = CACHE_SIZE;
         return this;
     }
+
+    public List<AssemblyEntity> runFilteredQuery(AliasFilters.Assembly filter) {
+        return new QueryUtils(entityManagerFactory).runFilteredQuery(filter);
+    }
+
+    private static class QueryUtils {
```

Comment:
all this would work but it's too manual. try query by example: https://docs.spring.io/spring-data/data-commons/docs/current/reference/html/#query-by-example

I think with that we can remove QueryUtils and AliasFilters, and we won't need to inject the entityManager